js-core/homeworks/homework-6/src/main.js

```diff
@@ -0,0 +1,503 @@
+  name: 'Albina',
+};
+
+let javaScript = {
+  html: 'JavaScript',
+};
+
+function countLetterA(str) {
+    let counterA = 0;
+    let strTransformToArray = str.split('');
+    strTransformToArray.forEach(function(value, index, arr) {
+        if(value === 'a') {
+            counterA += 1;
+        }
+    });
+    return counterA;
+}
+
+console.log(countLetterA(randomString)); // 4
+console.log(countLetterA(user.name + javaScript.html)); // 3
+
+// ! ! ! ! ! ! ! ! ! ! !
+/*
+ *
+ * TASK 2
+ * Вам дано предложение, вам необходимо перевернуть каждое
+ * слово в предложении в обратном порядке
+ *
+ * */
+
+function reverseEachWord() {
+}
+
+function reverseEachWord(str) {
+    let newArray = [];
+    let strTransformToArray = str.split(' ');
+    strTransformToArray.forEach(function(value, index, arr) {
+        let reversingEachWord = value.split('').reverse().join('');
+        newArray.push(reversingEachWord);
+    });
+    let newArrayTransformToString = newArray.join(' ');
+    return newArrayTransformToString;
+}
+
+
+console.log(reverseEachWord('You don\'t have to do anything special to begin using the DOM. Different browsers have different implementations of the DOM'));
+// uoY t'nod evah ot od gnihtyna laiceps ot nigeb gnisu eht .MOD tnereffiD sresworb evah tnereffid snoitatnemelpmi fo eht MOD
+
+console.log(reverseEachWord('The Document Object Model (DOM) is a programming interface for HTML and XML documents'));
+// ehT tnemucoD tcejbO ledoM )MOD( si a gnimmargorp ecafretni rof LMTH dna LMX stnemucod
+
+/*
+ * TASK 3
+ * Добавьте в функцию reverseEachWord второй параметр,
+ * данный параметр булево, если true - тогда предложение так же
+ * переворачиваются в обратном порядке
+ * */
+
+function reverseEachWord2(str, boolean) {
+    if(boolean === true) {
+        let reverseStr = str.split(' ').reverse().join(' ');
+        return reverseEachWord(reverseStr);
+    } else {
+        return reverseEachWord(str);
+    }
+}
+
+console.log(reverseEachWord2('You don\'t have to do anything special to begin using the DOM. Different browsers have different implementations of the DOM', true));
+//MOD eht fo snoitatnemelpmi tnereffid evah sresworb tnereffiD .MOD eht gnisu nigeb ot laiceps gnihtyna od ot evah t'nod uoY
+console.log(reverseEachWord2('The Document Object Model (DOM) is a programming interface for HTML and XML documents', true));
+// stnemucod LMX dna LMTH rof ecafretni gnimmargorp a si )MOD( ledoM tcejbO tnemucoD ehT
+
+console.log(reverseEachWord2('Hi my Name is', false));
+// iH ym emaN si
+
+/* TASK 3
+ * Посчитайте сколько одинаковых слов в предложении.
+ * Функция должна возвращать объект у которого свойства
+ * это повторяющиеся слово,
+ * а значение это количество повторений
+ * */
+
+// Both - Java - and - Java - Script - is - programming - and - programming - OOPBased
+
+function wordCounter(sentence) {
+    const sentenceTransformToArray = sentence.split(' ');
+    const sortSentenceArray = sentenceTransformToArray.sort();
```

Comment:
You don't have to sort that array, it should not affect the final result, but if you use it has some performance degradation